--- a/mt_wifi/embedded/sta/sanity.c
+++ b/mt_wifi/embedded/sta/sanity.c
@@ -374,8 +374,14 @@ BOOLEAN PeerAssocRspSanity(
 			ft_mic_cont.rsnxe_len = pEid->Len + 2;
 #endif /* DOT11R_FT_SUPPORT */
 			/* Copy whole RSNXE context */
-			NdisMoveMemory(&ie_list->rsnxe_ie_len, pEid, pEid->Len + 2);
-			ie_list->rsnxe_ie_len = pEid->Len + 2;
+			/* check length overflow before memmove */
+			if ((pEid->Len + 2) <= MAX_LEN_OF_RSNXEIE) {
+				NdisMoveMemory(ie_list->rsnxe_ie, pEid, pEid->Len + 2);
+				ie_list->rsnxe_ie_len = pEid->Len + 2;
+			} else {
+				MTWF_LOG(DBG_CAT_CLIENT, DBG_SUBCAT_ALL, DBG_LVL_WARN,
+					("%s(): RSNXE IE length(%d) exceeds maximum\n", __func__, pEid->Len + 2));
+			}
 			break;
 
 
