--- a/drivers/net/ethernet/mediatek/mtk_hnat/hnat.h
+++ b/drivers/net/ethernet/mediatek/mtk_hnat/hnat.h
@@ -851,7 +851,13 @@ struct foe_entry {
 		struct hnat_ipv6_5t_route ipv6_5t_route;
 		struct hnat_ipv6_6rd ipv6_6rd;
 		struct hnat_ipv6_hnapt ipv6_hnapt;
-		u32 data[31];
+#if defined(CONFIG_MEDIATEK_NETSYS_V3)
+		/* 4 + 31 * 4 = 128 Bytes for NETSYS_V3 */
+        u32 data[31];
+#else
+		/* 4 + 23 * 4 = 96 Bytes for NETSYS_V2 */
+        u32 data[23];
+#endif
 	};
 };
 
@@ -882,7 +888,6 @@ struct foe_entry {
 #define MAX_PPE_CACHE_NUM	(32)
 #endif
 
-#define PPE_ENTRY_SIZE		(128)
 
 /* If the user wants to set skb->mark to prevent hardware acceleration
  * for the packet flow.

--- a/drivers/net/ethernet/mediatek/mtk_hnat/hnat_nf_hook.c
+++ b/drivers/net/ethernet/mediatek/mtk_hnat/hnat_nf_hook.c
@@ -1661,7 +1661,7 @@ static int hnat_foe_entry_commit(struct
 	 * prevent racing with hardware accesses.
 	 */
 	memcpy(&foe->data, &entry->data,
-	       PPE_ENTRY_SIZE - sizeof(foe->info_blk1));
+	       sizeof(struct foe_entry) - sizeof(foe->info_blk1));
 	/* We must ensure all info has been updated before set to hw */
 	wmb();
 
